src/lm_buddy/jobs/_entrypoints/ragas.py

```diff
@@ -0,0 +1,105 @@
+from pathlib import Path
+
+from datasets import load_dataset
+from langchain_community.embeddings import HuggingFaceEmbeddings
+from langchain_openai import ChatOpenAI
+from lmbuddy.integrations.huggingface import HuggingFaceAssetLoader
+from ragas import evaluate
+from ragas.metrics import answer_relevancy, context_precision, context_recall, faithfulness
+
+from lm_buddy.integrations.wandb import artifact_loader
+from lm_buddy.integrations.wandb.artifact_config import WandbArtifactConfig
+from lm_buddy.integrations.wandb.artifact_utils import (
+    ArtifactType,
+    build_directory_artifact,
+    default_artifact_name,
+)
+from lm_buddy.integrations.wandb.run_utils import wandb_init_from_config
+from lm_buddy.jobs.common import EvaluationResult, LMBuddyJobType
+from lm_buddy.jobs.configs import RagasJobConfig
+
+RAGAS_METRICS_MAP = {
+    "faithfulness": faithfulness,
+    "answer_relevancy": answer_relevancy,
+    "context_recall": context_recall,
+    "context_precision": context_precision,
+}
+
+
+def run_eval(config: RagasJobConfig, artifact_loader: artifact_loader) -> Path:
+    # load dataset from W&B artifact
+    hf_loader = HuggingFaceAssetLoader(artifact_loader)
+    evaluation_dataset = hf_loader.load_dataset(config.dataset)
+
+    # ragas custom model args
+    ragas_args = {}
+
+    # load embedding model
+    ragas_args["embeddings"] = HuggingFaceEmbeddings(model_name=config.evaluation.embedding_model)
+
+    # configure ragas to point to vllm instance for generation
+    ragas_args["llm"] = ChatOpenAI(
+        model=config.ragas_inference_server.engine,
+        openai_api_key=config.evaluation.openai_api_key,
+        # get api endpoint from environment variable
+        openai_api_base=config.judge.inference.base_url,
+        max_tokens=config.judge.max_tokens,
+        temperature=config.judge.temperature,
+        top_k=config.judge.top_k,
+    )
+
+    result = evaluate(
+        dataset=evaluation_dataset,
+        metrics=RAGAS_METRICS_MAP[config.evaluation.metrics],
+        **ragas_args,
+    )
+    result_df = result.to_pandas()
+
+    # open the output file for writing and iterate on samples
+    tracking_name = config.tracking.name if config.tracking is not None else "output.json"
+    output_fname = Path(config.evaluation.output_folder) / tracking_name
+    result_df.to_json(output_fname)
+
+    # convert plain json dataset in HF format
+    output_dataset_path = Path(config.evaluation.output_folder) / "hf" / tracking_name
+    ds = load_dataset("json", data_files=str(output_fname), split="train")
+    ds.save_to_disk(output_dataset_path)
+
+    return output_dataset_path
+
+
+def run_ragas(config: RagasJobConfig, artifact_loader: artifact_loader) -> EvaluationResult:
+    # Run eval and store output in local filename
```

Comment:
might want to change the comment since we also have a `run_eval` function, to be clearer on what this does